src/test-derivatives.cpp

```diff
@@ -0,0 +1,94 @@
+#include <Rcpp.h>
+#define INCLUDE_RCPP
+#include "utils.h"
+#include "covariance.h"
+#include "chol_derivative.h"
+#include "testthat-helpers.h"
+#include <iostream>
+
+context("cho_jacobian") {
+  test_that("cho_jacobian works as expected") {
+    chol_jacobian chol_jac_obj(2, "ar1");
+    vector<double> theta {{1.0, 1.0}};
+    vector<double> result = chol_jac_obj(theta);
+    vector<double> expected(8);
+    expected << 2.718282, 1.922116, 0, 1.922116, 0.0, 0.9610578, 0.0, -0.9610578;
```

Comment:
do we know these values from R calculations? or is it a snapshot test?

Reply:
this is obtained from the previous design code using R. Shall we define cpp functions to obtain the derivatives, or just use something like this ? (using R to get the derivatives is easy and function is almost ready in the doc)

Reply:
Ok this is fine then, just add a comment where the numbers come from

Reply:
updated